compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/TilingInterfaceImpl.cpp

```diff
@@ -1637,44 +1670,70 @@ AttentionOp::getTiledImplementation(OpBuilder &builder,
 
   Location loc = getLoc();
   auto one = builder.getIndexAttr(1);
-  auto zero = builder.getIndexAttr(0);
 
-  SmallVector<OpFoldResult> queryOutputOffsets(getQueryRank(), zero);
-  SmallVector<OpFoldResult> queryOutputStrides(getQueryRank(), one);
-  ArrayRef<int64_t> queryShape = getQueryType().getShape();
-  SmallVector<OpFoldResult> queryOutputSizes =
-      getAsOpFoldResult(builder.getIndexArrayAttr(queryShape));
-  for (auto [idx, info] : llvm::enumerate(llvm::zip_equal(offsets, sizes))) {
-    queryOutputOffsets[idx] = std::get<0>(info);
-    queryOutputSizes[idx] = std::get<1>(info);
-  }
+  auto tileValue = [&](Value val, AffineMap indexingMap)
+      -> std::tuple<SmallVector<OpFoldResult>, SmallVector<OpFoldResult>,
+                    SmallVector<OpFoldResult>> {
+    assert(indexingMap.isProjectedPermutation() &&
+           "Indexing map should be a projected permutation");
+    SmallVector<OpFoldResult> outputOffsets;
+    SmallVector<OpFoldResult> outputSizes;
+    SmallVector<OpFoldResult> outputStrides(indexingMap.getNumResults(), one);
+    for (AffineExpr dimExpr : indexingMap.getResults()) {
+      auto dim = cast<AffineDimExpr>(dimExpr);
+      outputOffsets.push_back(offsets[dim.getPosition()]);
+      outputSizes.push_back(sizes[dim.getPosition()]);
+    }
+    return {outputOffsets, outputSizes, outputStrides};
```

Comment:
nit: use `std::make_tuple()`

Reply:
I'm not sure about this. I prefer this since the tuple type is already explicitly typed out in the function definition. So there is no ambiguity. Can change if there is a preference.